src/Subscription.php

```diff
@@ -385,19 +428,19 @@ public function swap($plan, $options = [])
 
         $subscription->save();
 
-        try {
-            $this->user->invoice(['subscription' => $subscription->id]);
-        } catch (StripeCard $exception) {
-            // When the payment for the plan swap fails, we continue to let the user swap to the
-            // new plan. This is because Stripe may attempt to retry the payment later on. If
-            // all attempts to collect payment fail, webhooks will handle any update to it.
-        }
-
         $this->fill([
             'stripe_plan' => $plan,
             'ends_at' => null,
         ])->save();
 
+        try {
+            $this->user->invoice(['subscription' => $subscription->id]);
+        } catch (IncompletePayment $exception) {
+            $this->markAsIncomplete();
+
+            throw $exception;
```

Comment:
This is maybe not totally the expected logic I would want to see. Imagine a user has a monthly Forge plan and 15 days into their subscription they decide to swap to yearly. If the payment is incomplete, this marks the whole subscription as incomplete and the user would no longer be considered as having an "active" subscription, even though they have already paid for the whole month.

Reply:
Hmm, you're correct. I hadn't considered that scenario. In that regard I think we should also update the webhooks since it will now always place a subscription in an incomplete state when its Stripe status changes to `incomplete` or `incomplete_expired`. I'll check into this and make some changes.

Reply:
@taylorotwell I've checked into this and can confirm that no exception is being thrown when downgrading or when proration applies. Stripe doesn't throws an exception in this case. Stripe will mark the subscription as "past_due" in this case which means Stripe will try to re-charge the card in the intervals set in the dashboard. When a "next action" is required for the payment the `invoice.payment_action_required` is fired which the email setting in this PR is for. 

In the other situations during a card error or "next action" payment status when you swap to a plan and need to immediately pay for an invoice the subscription is put into an incomplete state, just like when creating one. But I believe that's wanted, right?